### PR TITLE
GNN Facebook Page Classification

### PR DESCRIPTION
**Pull request by OllieV344**
This pull request implements a multi-layer Graph Neural Network (GNN) for semi-supervised node classification on the Facebook Large Page-Page Network dataset. The model combines configurable graph convolution layers with residual connections and batch normalization to effectively classify Facebook pages into categories using both network structure and node features.

Files Modified/Added
recognition/
└── gnn_facebook_4743209/
    ├── dataset.py      # Dataset loading and preprocessing
    ├── modules.py      # Model architecture implementation
    ├── train.py        # Training pipeline
    ├── predict.py      # Inference and analysis
    ├── requirements.txt # Dependencies
    └── README.md       # Documentation

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**
GNN FB Page-Page Network dataset ---- Normal Difficulty
| Category	| | Marks | Comments |	
| :---  | :---: | :---: | :---  |		
| Algorithm solves the problem 		| 5 | 4.5 | Some overfitting seen|
| Implementation functions as intended	| 3 | 3 | 6 layer GCN, torch_geometric/GCNConv |
| Good design 					| 1 | 1 | Modular, Reusable |
| Commenting 					| 1 | 1 | Meaningful docstrings and comments |
| Algorithm above Normal Difficulty 	| 5 | 5 | |
| Algorithm is Hard difficulty 			| 5 | 0 | Normal Difficulty |
| Section IV : Max mark 15 from 20		|   | **14.5**   | |

**TSNE / UMAP:** 
Good, though would have expected to see a bit more separation at the reported accuracy.
- Discussion: Goo.

**Suggestions:**
Using 6 layers is probably too high, and possibly the cause for the overfitting. Did you try 2 or 3 layers?

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some overfitting seen, possibly due to 6 layers.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. GNN Task.</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. Commits contain previous GAN work, and only 2 days of GCN. This is fine.</td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Add some info about the specific implementation here, rather than a file list.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-8</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Oliver Vinton
